### PR TITLE
Add ability to connect to integrations from the SDK

### PR DESCRIPTION
**Pull request by kenxu95**
## Describe your changes and why you are making these changes
Duplicated some structs that aqueduct_executor also owned. Will add a tech debt task to deduplicate later.

## Related issue number (if any)
ENG-1837

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: sdk/aqueduct/integrations/connect_config.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    database: str
+    host: str
+    port: str
+
+
+class SQLiteConfig(BaseConnectionConfig):
+    database: str
+
+
+IntegrationConfig = Union[
```
Comment (hsubbaraj-spiral) [inline sdk/aqueduct/integrations/connect_config.py:116]: Are we allowing users to connect compute integrations (K8s, Lambda, etc.) through the SDK?

File: sdk/aqueduct/integrations/connect_config.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+from typing import Dict, Optional, Union
+
+from aqueduct.constants.enums import MetaEnum, ServiceType
+from aqueduct.error import InternalAqueductError
+from pydantic import BaseModel, Extra, Field
+
+"""Copied mostly over from `aqueduct_executor/operators/connectors/data/config.py` for now, please keep them in sync."""
+
+
+class BaseConnectionConfig(BaseModel):
```
Comment (hsubbaraj-spiral) [inline sdk/aqueduct/integrations/connect_config.py:11]: What is the end goal for these structs? Should we move them to a `shared` directory in the sdk and have `aqueduct_executor` refer to these?

**Review by hsubbaraj-spiral — CHANGES_REQUESTED** [review]

File: sdk/aqueduct/integrations/connect_config.py (hunk 1 above)
Comment (kenxu95) [reply sdk/aqueduct/integrations/connect_config.py:116]: yeah I think so - I figured we could add them here as we go.

File: sdk/aqueduct/integrations/connect_config.py (hunk 2 above)
Comment (kenxu95) [reply sdk/aqueduct/integrations/connect_config.py:11]: Yeah I would like to get it shared between them the same way our serialization logic is. But there are a couple discrepancies that were introduced to get this working on the SDK side, so to keep this well-scoped, I figured we could duplicate them for now to get the feature out and have a cleanup task for merging them.

Comment (kenxu95) [reply sdk/aqueduct/integrations/connect_config.py:11]: https://linear.app/aqueducthq/issue/ENG-2235/share-integrationconfig-between-sdk-and-executor

**Review by hsubbaraj-spiral — APPROVED** [review]

**Pull request merged**